drgn/helpers/memory.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+SYMBOL_KIND_STR = {
+    SymbolKind.UNKNOWN: "symbol",
+    SymbolKind.OBJECT: "symbol object",
+    SymbolKind.FUNC: "symbol function",
+    SymbolKind.SECTION: "symbol section",
+    SymbolKind.FILE: "symbol file",
+    SymbolKind.COMMON: "symbol common",
+    SymbolKind.TLS: "symbol tls",
+    SymbolKind.IFUNC: "symbol ifunc",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """
```

Comment:
After your last PR made me realize how much it sucked to duplicate the documentation, I made it so that's not necessary. See https://github.com/osandov/drgn/commit/5b6a8c27a9586052b3f81d768880e4c39ac3414e; you can replace the docstring in the first overload with an empty one (`""""""`) and then put all of the documentation in the second one.

Reply:
Oh wow that's much better, fixed!